stage0/README.md

```diff
@@ -4,14 +4,40 @@
 <h1><picture><source media="(prefers-color-scheme: dark)" srcset="/docs/oak-logo/svgs/oak-stage-0-negative-colour.svg?sanitize=true"><source media="(prefers-color-scheme: light)" srcset="/docs/oak-logo/svgs/oak-stage-0.svg?sanitize=true"><img alt="Project Oak Logo" src="/docs/oak-logo/svgs/oak-stage-0.svg?sanitize=true"></picture></h1>
 <!-- Oak Logo End -->
 
-This is a minimal firmware for use with QEMU and compatible VMMs that puts the
-CPU into the desired state, enters long mode, retrieves required configuration
-information (E820 memory layout and ACPI tables) and jumps into the kernel entry
-point.
+Oak stage0 is a purposefully minimal firmware for virtual machines that can boot
+64-bit Linux(-compatible) ELF kernels.
 
-The E820 memory map and ACPI tables are read using the
-[QEMU Firmware Configuration (fw_cfg)](https://www.qemu.org/docs/master/specs/fw_cfg.html)
-device.
+The motivating example for having a minimal firmware is secure virtual machines,
+such as virtual machines running under AMD SEV-SNP or Intel TDX, where it is
+beneficial to minimize the amount of code that needs to be trusted (that is,
+running inside the virtual machine).
+
+To that end, stage0 intentionally tries to limit itself to the minimum amount of
+actions needed to set up the machine, preferring to do as much work in
+memory-safe Rust as possible. Unfortunately we're still dealing with low-level
+machine state here, so we've had to resort to some assembly code and some
+`unsafe` code.
+
+## Features
+
+We target the QEMU `microvm` machine (and compatible VMMs), and support:
+
+- [QEMU `fw_cfg`](https://www.qemu.org/docs/master/specs/fw_cfg.html) device
+  (for obtaining memory information, ACPI tables etc)
+- serial port (for logging)
+- AMD SEV, SEV-ES and SEV-SNP (setting encrypted bit in the page tables and
+  validating guest physical memory)
+- loading and parsing ELF kernels
+- the 64-bit Linux boot protocol
```

Comment:
This is not strictly accurate, but I am not sure what the more accurate name is. The 64-bit boot protocol refers to jumping into the 64-bit entry point of the compressed Linux kernel image (vmlinuz or bzImage). We are jumping into the entry point of the extracted 64-bit ELF file.

In spite of the pedantic rant it might still be fine to use the term "64-bit Linux boot protocol", since we use the same boot parameters as is expected by the 64-bit boot protocol.

Reply:
Yes, that was my point: we also set up the zero page, and all the related structures, according to the Linux boot protocol. I'd say that's more important than the fact that we're jumping to the ELF entry point here.